src/main/java/gov/nj/innovation/customAwsIdp/lambda/GetSamlResponseHandler.java

```diff
@@ -0,0 +1,191 @@
+        }
+        if (rp.duration() == null || rp.duration().isBlank() || !DIGITS_PATTERN.matcher(rp.duration()).matches() ||
+                Integer.parseInt(rp.duration()) < 900 || Integer.parseInt(rp.duration()) > 43200) {
+            return createErrorReturnMap(Status.INPUT_ERROR,
+                    String.format("Invalid duration, must be an int between 900 and 43200, was %s", rp.duration()));
+        }
+
+        final AuthorizerContextDetails acd = extractAuthorizerDetailsFromInput(input);
+        if (acd.email() == null || acd.email().isBlank()) {
+            return createErrorReturnMap(Status.INPUT_ERROR,
+                    String.format("email for Cognito user session must have a non-empty value, had %s", acd.email()));
+        }
+        if (!acd.usersGroups().contains(rp.groupName())) {
+            return createErrorReturnMap(Status.INPUT_ERROR,
+                    String.format("User %s does not belong to a group named %s", acd.email(), rp.groupName()));
+        }
+
+        final CognitoGroupDescriptionMetadata ssoMetadata;
+        try {
+            ssoMetadata = CognitoGroupDescriptionMetadataExtractor.extract(REGION, rp.groupName(), USER_POOL);
+        } catch (RuntimeException e) {
+            return createErrorReturnMap(Status.SYSTEM_ERROR,
+                    String.format("Error trying to extract metadata from Group %s in UserPool %s: %s",
+                            rp.groupName(), USER_POOL, e.getMessage()));
+        }
+
+        final KeyConstants keyConstants;
+        try {
+            keyConstants = new KeyConstants();
+        } catch (NullPointerException | NumberFormatException e) {
+            return createErrorReturnMap(Status.SYSTEM_ERROR,
+                    String.format("KeyConstants threw an exception: %s.\nNOTE: Please check that the " +
+                            "key-secrets are correct in AWS Systems Manager", e.getMessage()));
+        }
+
+        try {
+            final KeysWrapper keys = new KeysWrapper(keyConstants);
+            final SamlGenerator generator = new SamlGenerator(acd.email(), ssoMetadata.ssoRole(), rp.duration(), keys);
+            final String encodedSamlResponse = generator.getBase64SamlResponse();
+            logger.info(String.format("Successfully generated SAML Response for user %s with role %s and duration %s",
+                    acd.email(), ssoMetadata.ssoRole(), rp.duration()));
+            return createReturnMap(Status.SUCCESS, encodedSamlResponse, ssoMetadata.relayState(), null);
+
+        } catch (Exception e) {
+            return createErrorReturnMap(Status.SYSTEM_ERROR,
+                    String.format("Error in handler: user %s, ssoRole %s, duration %s; error message: %s",
+                            acd.email(), ssoMetadata.ssoRole(), rp.duration(), e.getMessage()));
+        }
+    }
+
+    /* Casting the Object results of input.get() leads to unchecked warnings. Ignore them for now. */
+    @SuppressWarnings("unchecked")
```

Comment:
Wondering if `RequestHandler<Map<String, Object>` could be defined in `Handler` as `RequestHandler<Map<String, Map<String, String>>`? Would that simplify things?

Reply:
I think we shouldn't do that because the type isn't so easily nailed down.

If we wanted to be very correct, we could invent a class like `InputData` which would have the keys and their values and nested value types defined within, but that seems kind of like overkill for now.

But if you check the comment on the class, you see that sometimes the Object-value is a `Map<String, String>` (in the case of `pathParameters` and `queryStringParameters`, sometimes it is `Map<String, Map<String, Map<String, Map<String, String>>>>` (in the case of `requestContext.authorizer.jwt.claims`), and it may be other things for parts of the input I am not using here.